bot/utils/transformData.ts

```diff
@@ -0,0 +1,96 @@
+import { poolpairs, stats } from '@defichain/whale-api-client';
+import { BigNumber } from 'bignumber.js';
+
+import { StateRelayer } from '../../generated';
+import { MasterNodeData, PairData, VaultData } from './types';
+
+
+const DECIMALS = 18;
+const DENOMINATION = 'USDT';
+
+type AsyncReturnType<T extends (...args: any) => any> = T extends (...args: any) => Promise<infer R> ? R : any;
+type StatsData = AsyncReturnType<typeof stats.Stats.prototype.get>;
+type PoolPairData = AsyncReturnType<typeof poolpairs.PoolPairs.prototype.list>;
+type DexPriceResult = AsyncReturnType<typeof poolpairs.PoolPairs.prototype.listDexPrices>;
+
+const transformToBigInt = (str: string, decimals: number): bigint =>
+  BigInt(new BigNumber(str).multipliedBy(new BigNumber('10').pow(decimals)).toFixed(0, 1));
+
+export function transformDataVault(statsData: StatsData): VaultData {
+  const totalLoanValue = statsData.loan.value.loan;
+  const totalCollateralValue = statsData.loan.value.collateral;
+  const dataVault: VaultData = {
+    noOfVaultsNoDecimals: transformToBigInt(statsData.loan.count.openVaults.toString(), 0),
+    totalLoanValue: transformToBigInt(totalLoanValue.toString(), DECIMALS),
+    totalCollateralValue: transformToBigInt(totalCollateralValue.toString(), DECIMALS),
+    totalCollateralizationRatio: transformToBigInt(
+      ((totalCollateralValue / totalLoanValue) * 100).toFixed(3).toString(),
+      DECIMALS,
+    ),
+    activeAuctionsNoDecimals:  transformToBigInt(statsData.loan.count.openAuctions.toString(), 0)
+  };
+  return dataVault;
+}
+
+export function transformDataMasternode(statsData: StatsData): MasterNodeData {
+  const dataMasterNode: MasterNodeData = {
+    totalValueLockedInMasterNodes: transformToBigInt(statsData.tvl.masternodes.toString(), DECIMALS),
+    zeroYearLockedNoDecimals: transformToBigInt(statsData.masternodes.locked.find(i => i.weeks === 0)!.count.toString(), 0),
+    fiveYearLockedNoDecimals: transformToBigInt(statsData.masternodes.locked.find(i => i.weeks === 260)!.count.toString(), 0),
+    tenYearLockedNoDecimals:  transformToBigInt(statsData.masternodes.locked.find(i => i.weeks === 520)!.count.toString(), 0)
+  }
+  return dataMasterNode;
+}
+
+export function tranformPairData(
+  rawPoolPairData: Array<poolpairs.PoolPairData>,
+  statsData: StatsData,
+  dexPriceData: DexPriceResult,
+): {
+  dex: string[];
+  dexInfo: StateRelayer.DEXInfoStruct[];
+  totalValueLocked: BigInt;
```

Comment:
I think this should be `bigint` type instead
```suggestion
  totalValueLocked: bigint;
```